torch/optim/swa_utils.py

```diff
@@ -242,15 +238,8 @@ def update_parameters(self, model: Module):
 
         if self.n_averaged > 0:
             if self.multi_avg_fn is not None or self.avg_fn is None:
-                grouped_tensors = _group_tensors_by_device_and_dtype(
-                    cast(TensorListList, [self_param_detached, model_param_detached])
-                )
-                grouped_tensors = cast(
-                    Dict[
-                        Tuple[torch.device, torch.dtype],
-                        Tuple[List[List[Tensor]], Indices],
-                    ],
-                    grouped_tensors,
+                grouped_tensors = Optimizer._group_tensors_by_device_and_dtype(
```

Comment:
Why the switch to use the optimizer one here?

Reply:
The optimizer one internally calls the original function and also supports compiling. It doesn't need to perform a lot of type casting because it is ignored in the optimizer one. Is this not suitable here?"

Reply:
eh...the one in optimizer is a special case where it gets skipped in compile world..I'd rather not use it here if not necessary yet.

Reply:
Okay, I'll roll it back. Thanks for explaining this.